xc/xenops_server_xen.ml

```diff
@@ -2855,6 +2871,20 @@ module Actions = struct
 						None in
 				Opt.iter (fun x -> Updates.add x internal_updates) update in
 
+		let fire_event_on_qemu domid signal =
+			let d = int_of_string domid in
+			let open Xenstore_watch in
+			if not(IntMap.mem d domains)
+			then debug "Ignoring qemu-pid-signal watch on shutdown domain %d" d
+			else
```

Comment:
You need `begin..end` here, otherwise only the following debug line is part of the conditional... ⚠️